src/app/core/pending-changes.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { CanDeactivate } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+export interface ComponentCanDeactivate {
+  canDeactivate: () => boolean | Observable<boolean>;
+}
+
+
+@Injectable()
+export class PendingChangesGuard implements CanDeactivate<ComponentCanDeactivate> {
+  canDeactivate(component: ComponentCanDeactivate): boolean | Observable<boolean> {
+    // if there are no pending changes, just allow deactivation; else confirm first
+    return component.canDeactivate() ?
+      true:
+      // NOTE: this warning message will only be shown when navigating elsewhere within your angular app;
+      // when navigating away from your angular app, the browser will show a generic warning message
+      confirm('WARNING: You have unsaved changes. Press Cancel to go back and save these changes, or OK to lose these changes.');
```

Comment:
Wording doesn't match our service
```suggestion
      confirm('WARNING: Your message has not been sent yet. Press Cancel to go back to complete and send your message, or OK to discard your message.');
```